### ISSUE TITLE
feature request: transform email

### ISSUE THREAD

**Issue by michaelcadilhac** [marlam/mpop-mirror#12]
It would be nice if mpop could, on the fly, transform (headers of) emails.  Something like a `transform prog` option that runs `prog` with the email on stdin, and uses the stdout as the transformed email. 

**Rationale:** I'm fetching emails from an Outlook server, and storing them on a private server (maildir).  Gmail is then accessing my private server and fetching these.  Problematically, Outlook adds "Received" headers, and Gmail complains that Outlook is not a permitted sender for whichever address the email originated from.  I'd like to "sanitize" the email by flushing the "Received" headers from Outlook.

**Comment by marlam** [comment]
You can do that with `delivery mda /path/to/my-script`. The script reads the mail, optionally transforms it in some way, and then puts it into a maildir or mbox file (or uses a program such as `procmail` to do that last step).

**Comment by michaelcadilhac** [comment]
Aah, of course!  Thanks Martin.  This is not _exactly_ the same, since I have to rewire the maildir/mbox/exchange with a different tool rather than just use mpop, but this does work indeed.  Feel free to close this feature request if you think that it'd be redundant.

**Comment by marlam** [comment]
I agree that it would be nice to have a `transform` or `process` command that filters the mail before it goes to the delivery stage. I'll keep this issue open as a reminder for later.

**Comment by marlam** [comment]
It's been a while, but there is progress.

mpop can now act as a mail delivery agent (MDA): when given the `--mda` option, it reads a mail from standard input, and delivers it according to the `--delivery` option, just as it would with the `delivery` command.

So now mpop can be its own MDA:
```
# two ways of getting mail into ~/maildir:
delivery maildir ~/maildir
delivery mda "mpop --mda --delivery=maildir,~/maildir"
```

This means you can now easily insert filter scripts:
```
delivery mda "my-filter.py | mpop --mda --delivery=maildir,~/maildir"
```

This change is now ready for testing (41e626a838bfe56359024d0f8cfd4798d66a465d).

**Comment by michaelcadilhac** [comment]
This has been running flawlessly for the past few days, thanks @marlam !
